crates/static-analysis-kernel/src/analysis/ddsa_lib/context/file.rs

```diff
@@ -2,11 +2,27 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2024 Datadog, Inc.
 
+use crate::analysis::ddsa_lib::context::file_go::FileContextGo;
+
 #[derive(Debug, Default)]
 pub struct FileContext {
     // Supported file contexts:
+    go: Option<FileContextGo>,
 }
```

Comment:
We had previous conversation about making it an `enum` which we did? Why did it came back as a `struct`?